pytest_httpx/_httpx_mock.py

```diff
@@ -257,8 +278,12 @@ def _explain_that_no_response_was_found(self, request: httpx.Request) -> str:
             request_description += f" with {present_headers} headers"
             if expect_body:
                 request_description += f" and {request.read()} body"
+            elif expect_json:
+                request_description += f" and {request.read().decode('utf-8')} body"
         elif expect_body:
             request_description += f" with {request.read()} body"
+        elif expect_json:
+            request_description += f" with {request.read().decode('utf-8')} body"
```

Comment:
I dont think there is added value for this here, we should instead update the expect_body condition to be True in such a case

Reply:
The only difference indeed would be difference between a binary string being displayed vs a normal string. But since its json, it doesn't really matter indeed.